tools/fmt/fmt.rs

```diff
@@ -137,13 +145,18 @@ fn fold(
     }
 }
 
+enum SyntaxMatch {
```

Comment:
Added this enum to be able to distinguish between what we found in cases where we multiple syntaxkinds can come next.